src/custom-element.js

```diff
@@ -976,6 +996,9 @@ function createBaseCustomElementClass(win) {
      * @private @final @this {!Element}
      */
     tryUpgrade_() {
+      if (this.isAwaitingSize_()) {
```

Comment:
I think we need to continue upgrading right away. Notice upgrade doesn't mean layout, but rather initial config steps and possible build.